src/main/java/org/opensearch/security/user/UserService.java

```diff
@@ -314,4 +317,49 @@ public static void saveAndUpdateConfigs(
             throw ExceptionsHelper.convertToOpenSearchException(e);
         }
     }
+
+    /**
+     * Removes accounts that are not of the requested type from the SecurityDynamicConfiguration object passed.
+     *
+     * Accounts with the 'service' attribute set to true, are considered of type 'service'.
+     * Accounts with the 'service' attribute set to false or without the 'service' attribute, are considered of type 'internal'.
+     *
+     * @param configuration SecurityDynamicConfiguration object containing all accounts
+     * @param requestedAccountType The type of account to be kept. Should be "service" or "internal"
+     *
+     */
+    public void includeAccountsIfType(SecurityDynamicConfiguration<?> configuration, AccountType requestedAccountType)
+        throws UserServiceException {
```

Comment:
I don't think this exception is required, exceptions as part of normal flow of control are frowned upon, lets removed https://stackoverflow.com/questions/729379/why-not-use-exceptions-as-regular-flow-of-control